publish.sh

```diff
@@ -44,55 +47,81 @@ function run_provision_manager() {
-
-    cluster-provision/gocli/build/cli provision --phases k8s cluster-provision/k8s/$i --slim
-    ${CRI_BIN} tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim
+  for i in "${IMAGES_TO_BUILD[@]}"; do
+    if [ $ARCH == "amd64" ]; then
+      echo "INFO: building $i"
+      cluster-provision/gocli/build/cli provision --phases k8s cluster-provision/k8s/$i
+      ${CRI_BIN} tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}
+
+      cluster-provision/gocli/build/cli provision --phases k8s cluster-provision/k8s/$i --slim
+      ${CRI_BIN} tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim
+    elif [ $ARCH == "s390x" ] && [ $i == "1.30" ]; then
+      echo "INFO: building $i slim"
+      cluster-provision/gocli/build/cli provision --phases k8s cluster-provision/k8s/$i --slim
+      ${CRI_BIN} tag ${TARGET_REPO}/k8s-$i ${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}
+    fi
   done
 }
 
 function push_node_base_image() {
-  TARGET_IMAGE="${TARGET_REPO}/centos9:${KUBEVIRTCI_TAG}"
+  if [ $ARCH == "amd64" ]; then
+    TARGET_IMAGE="${TARGET_REPO}/centos9:${KUBEVIRTCI_TAG}"
+  else
+    TARGET_IMAGE="${TARGET_REPO}/centos9:${KUBEVIRTCI_TAG}-${ARCH}"
+  fi
   podman tag ${TARGET_REPO}/centos9-base:latest ${TARGET_IMAGE}
   echo "INFO: push $TARGET_IMAGE"
   podman push ${TARGET_IMAGE}
-  echo ${TARGET_IMAGE} > cluster-provision/k8s/base-image
 }
 
 function push_cluster_images() {
-  for i in ${IMAGES_TO_BUILD[@]}; do
-    echo "INFO: push $i"
-    TARGET_IMAGE="${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
-    podman push "$TARGET_IMAGE"
-
-    TARGET_IMAGE="${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
-    podman push "$TARGET_IMAGE"
+  for i in "${IMAGES_TO_BUILD[@]}"; do
+    if [ $ARCH == "amd64" ]; then
+      echo "INFO: push $i"
+      TARGET_IMAGE="${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
+      podman push "$TARGET_IMAGE"
+
+      TARGET_IMAGE="${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
+      podman push "$TARGET_IMAGE"
+    elif [ $ARCH == "s390x" ] && [ $i == "1.30" ]; then
+      echo "INFO: push $i slim"
+      TARGET_IMAGE="${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}"
+      podman push "$TARGET_IMAGE"
+    fi
   done
 
   # images that the change doesn't affect can be retagged from previous tag
   for i in ${IMAGES_TO_RETAG[@]}; do
     echo "INFO: retagging $i (previous tag $PREV_KUBEVIRTCI_TAG)"
-    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
-    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
+    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-${ARCH}"
+    skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}"
```

Comment:
Thanks for pointing this.

I think this can be fixed as below - When this script run on x86 first, we are re-tagging only multi-arch manifest. Next, when run on s390x, s390x-arch image will be retagged.
```
  for i in ${IMAGES_TO_RETAG[@]}; do
    echo "INFO: retagging $i (previous tag $PREV_KUBEVIRTCI_TAG)"
    if [ $ARCH == "amd64" ]; then 
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}"
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim"
    elif [ $ARCH == "s390x" ] && [ $i == "1.30" ]; then
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-${ARCH}"
      skopeo copy "docker://${TARGET_REPO}/k8s-$i:${PREV_KUBEVIRTCI_TAG}-slim-${ARCH}" "docker://${TARGET_REPO}/k8s-$i:${KUBEVIRTCI_TAG}-slim-${ARCH}"
    fi
  done
```